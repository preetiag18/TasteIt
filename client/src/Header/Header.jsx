import React from "react";
import { NavLink } from "react-router-dom";

import Nav from "../Nav/Nav";

import classes from "./Header.module.css";

const Header = () => {
  return (
    <header className={classes.header}>
      <NavLink className={classes.title} to="/">
        <h2>TasteIt</h2>
      </NavLink>
      <Nav />
    </header>
  );
};

export default Header;
