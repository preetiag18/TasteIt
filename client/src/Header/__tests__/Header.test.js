import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Header from "../Header";
import { BrowserRouter } from "react-router-dom";

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
