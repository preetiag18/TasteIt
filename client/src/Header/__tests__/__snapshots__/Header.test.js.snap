// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <a
    aria-current="page"
    className="title active"
    href="/"
    onClick={[Function]}
  >
    <h2>
      TasteIt
    </h2>
  </a>
  <nav>
    <button
      className="mob_button"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuRoundedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 18h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zm0-5h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1s.45 1 1 1zM3 7c0 .55.45 1 1 1h16c.55 0 1-.45 1-1s-.45-1-1-1H4c-.55 0-1 .45-1 1z"
        />
      </svg>
    </button>
    <ul
      className="hideUl"
    >
      <li
        className=""
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className=""
      >
        <a
          className=""
          href="/recipes"
          onClick={[Function]}
        >
          Recipes
        </a>
      </li>
      <li
        className=""
      >
        <a
          className=""
          href="/addnewrecipes"
          onClick={[Function]}
        >
          Add new recipes
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
